From 9f38ba7d9eff24b4c32760d0bb4b9f653a934203 Mon Sep 17 00:00:00 2001
From: Devajith Valaparambil Sreeramaswamy
 <devajith.valaparambil.sreeramaswamy@cern.ch>
Date: Thu, 31 Oct 2024 09:32:16 +0100
Subject: [PATCH 03/51] Revert "Disable validation of transent files."

This reverts commit fb0019d3392bd91b8be0b93da913e699dff36e2b.
---
 .../llvm-project/clang/lib/Serialization/ASTReader.cpp      | 6 +-----
 1 file changed, 1 insertion(+), 5 deletions(-)

diff --git a/interpreter/llvm-project/clang/lib/Serialization/ASTReader.cpp b/interpreter/llvm-project/clang/lib/Serialization/ASTReader.cpp
index cb2793b2b0..01720460d3 100644
--- a/interpreter/llvm-project/clang/lib/Serialization/ASTReader.cpp
+++ b/interpreter/llvm-project/clang/lib/Serialization/ASTReader.cpp
@@ -2543,12 +2543,8 @@ InputFile ASTReader::getInputFile(ModuleFile &F, unsigned ID, bool Complain) {
 
   // For an overridden file, create a virtual file with the stored
   // size/timestamp.
-  bool DisableValidation = bool(PP.getPreprocessorOpts().DisablePCHOrModuleValidation &
-                                DisableValidationForModuleKind::All);
-  if ((Overridden || Transient || SkipChecks) && (!File || DisableValidation)) {
+  if ((Overridden || Transient || SkipChecks) && !File)
     File = FileMgr.getVirtualFileRef(Filename, StoredSize, StoredTime);
-    Overridden = true;
-  }
 
   if (!File) {
     if (Complain) {
-- 
2.43.5

